client/src/components/SearchBar/SearchBar.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useEffect, useState } from "react";
+import { OutlinedInput, InputLabel, InputAdornment } from "@material-ui/core";
+import LocationOnIcon from "@material-ui/icons/LocationOn";
+import LocationOffIcon from "@material-ui/icons/LocationOff";
+import { isDebug } from "src/config";
+
+function SearchBar({
+  iconCallback,
+  onChangeCallback,
+  onEnterCallback,
+}: {
+  iconCallback?: (status: boolean) => void;
+  onChangeCallback?: (text: string) => void;
+  onEnterCallback?: (text: string) => void;
+}) {
+  const [gpsStatus, setGpsStatus] = useState(false);
+
+  const iconClickWrapper = () => {
+    setGpsStatus(!gpsStatus);
+  };
+
+  const dynamicChangeWrapper = (
```

Comment:
could provide a better name for the dynamicChangeWrapper? e.g. updateSearchResults
```suggestion
  const betterName = (
```

Reply:
renamed to updateTextWrapper (I try to keep '*Wrapper' as the real functionality is a given specification via `onChangeCallback`)